### DISCUSSION TITLE
Instructions on how to cherry pick a commit from another branch to current branch

### DISCUSSION THREAD

**Discussion by searleser97** [NeogitOrg/neogit#907] [Q&A] [answered]
Can someone help me to understand what is the workflow to cherry pick a commit from another branch to current branch ?

How do I get the list of commits from any branch ? I couldn't understand it

**Comment by CKolkey** [comment, marked as answer]
Hey! Sure thing.

1. From the status buffer (`:Neogit`), open the Log Popup (`l` by default)
2. Check out that branch that you want to pull the commits into
3. Hit `o` for `Log other`. This will bring up a list of branches.
4. Select the branch you want to harvest commit(s) from.
5. The Log Buffer will open up, showing commits for that branch.
6. You can now hit `A` to bring up the cherry pick popup, and choose either `Pick` or `Apply`.
	- You can check out `:h neogit_cherry_pick_popup` for details on that
	- You can also do a linewise-visual selection (`V`) to select multiple commits.

Hitting `<cr>` (enter) will open the commit view for a specific commit - the cherry pick popup works from there too.

Let me know if you have any other questions :)

**Reply by searleser97** [reply]
Is there a way to list the remote branches instead of local branches?

**Reply by CKolkey** [reply]
Think that was just an oversight on my part. Added it here: https://github.com/NeogitOrg/neogit/pull/908

**Reply by khaitranhq** [reply]
@CKolkey I see we can only select a single commit with `A` and multiple continuous commits with the linewise-visual selection mode but we can't select non-continuous commits. Is this right? Can I configure this plugin to use snack to cherry pick commit?

**Reply by CKolkey** [reply]
Yeah, thats right. One at a time, or a continuous range. There's not a way to integrate a picker like snacks to do commits piecemeal, but maybe there's a built in picker?

Maybe as an alternative you could make your own picker that then passes the commit hash's off to neogit's backend, if you really wanted to.

**Reply by sassanh** [reply]
What if I want to cherry-pick a dangling commit based on its hash? Can I enter the commit hash manually instead of finding it somewhere and pressing A on it or selecting it in log viewer?

**Reply by CKolkey** [reply]
You could do it slightly manually via `Q` - that lets you run any git command you like.

**Reply by sassanh** [reply]
Thanks, I didn't know about Q, it should be very handy for the edge cases.

The thing is one can always open a terminal with a shortcut they've set and run the desired git command. But to have things aligned with neogit's experience, and following the same UI codes, can't we have a picker input which shows the list of all commits in which one can filter them based on hash and commit message by typing in the input?

Please correct me if I'm wrong, but as far as I know one doesn't really need to cherry pick from history of the HEAD, because all those commits are already in the history of HEAD.
